modules/mobile/app/controllers/mobile/v0/appointments_controller.rb

```diff
@@ -81,19 +84,26 @@ def fetch_cached_or_service(validated_params)
 
         appointments.filter! { |appt| appt.is_pending == false } unless validated_params[:included]&.include?('pending')
         appointments.reverse! if validated_params[:reverse_sort]
+
         appointments
       end
 
       def paginate(appointments, validated_params)
         appointments = appointments.filter do |appointment|
           appointment.start_date_utc.between?(
-            validated_params[:start_date], validated_params[:end_date]
+            validated_params[:start_date].beginning_of_day, validated_params[:end_date].end_of_day
```

Comment:
I added this because I was testing with a one day date range and experiencing issues.

Reply:
do the start and end dates always formatted with just dates, no times? if so, could this also be utilized in the primary appointments call in `fetch_cached_or_service`?

Reply:
It's a good question, but `fetch_cached_or_service` has a totally different behavior. It always sets the begin date to at least a year ago and always sets the end date to at least a year from now: 
```
          appointments = appointments_proxy.get_appointments(
            start_date: [validated_params[:start_date], one_year_ago].min,
            end_date: [validated_params[:end_date], one_year_from_now].max
          )
```
It's combined with this method:
```
      def use_cache?(validated_params)
        # use cache if date range is within +/- 1 year and use_cache is true
        validated_params[:start_date] >= one_year_ago.iso8601 &&
          validated_params[:end_date] <= one_year_from_now.iso8601 && validated_params[:use_cache]
      end
```
to ensure that the cache is always accurate when requesting between those dates. Which is undoubtedly the large majority of requests. If we're outside of those dates, we don't use the cache. There might also be some slight edge cases there, but they're a lot less noticeable because of the wider date range. The issue that caused me to experience this bug is that I was using very narrow date ranges. Which meant that the controller was requesting two years of data, but then filtering out everything.

Reply:
All good points. I also missed that the `one_year_from_now` and `one_year_ago` methods were already using the `beginning_of_day`